[webkitscmpy] Linkify http urls in pull-requests

JonWBedard opened the pull request:
#### 389b2003930433799f541bfc058f7b1cb139847c
<pre>
[webkitscmpy] Linkify http urls in pull-requests
<a href="https://bugs.webkit.org/show_bug.cgi?id=230655">https://bugs.webkit.org/show_bug.cgi?id=230655</a>

Reviewed by NOBODY (OOPS!).

* Scripts/libraries/webkitscmpy/webkitscmpy/pull_request.py:
(PullRequest):
(PullRequest.escape_html): Escape any html and linkify http urls.
(PullRequest.unescape_html): Unescape escaped html and strip links from http urls.
(PullRequest.create_body): Use &lt;pre &gt; and escape html in commit messages, unless opted out.
(PullRequest.parse_body): Match both &lt;pre &gt; and ```, escape html if &lt;pre &gt; was used.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/bitbucket.py:
(BitBucket.PRGenerator.create): Opt out of linkifying http in comment body.
(BitBucket.PRGenerator.update): Ditto.
* Scripts/libraries/webkitscmpy/webkitscmpy/test/pull_request_unittest.py:
</pre>

graouts commented on Source/WebCore/platform/graphics/mac/GraphicsContext3DMac.mm:
```diff
@@ -342,10 +342,10 @@ static void setPixelFormat(Vector<CGLPixelFormatAttribute>& attribs, int colorBi
 {
-    [m_webGLLayer setBounds:CGRectMake(0, 0, width, height)];
+    [m_webGLLayer setBackingStoreSize:CGSizeMake(width, height)];
     [m_webGLLayer setOpaque:(m_internalColorFormat != GL_RGBA8)];
```
I suppose this should now be set directly in the WebGLLayer initializer or in `setBackingStoreSize`.

grorg replied:
What "this" do you mean?

graouts replied:
The call to `[m_webGLLayer setOpaque:(m_internalColorFormat != GL_RGBA8)];`

graouts commented on Source/WebCore/platform/graphics/mac/WebGLLayer.h:
```diff
@@ -49,6 +52,11 @@ namespace WebCore {
 
+#if PLATFORM(IOS)
+@property (readonly, nonatomic) CAEAGLLayer* glLayer;
+- (void)setBackingStoreSize:(CGSize)size;
```
I was wondering if I can have both of those in the same `#if PLATFORM(IOS)` block or if there needs to be one for the property and one for the method.

grorg replied:
I think it is fine in one block.

graouts commented on Source/WebCore/platform/graphics/mac/WebGLLayer.mm:
```diff
@@ -45,12 +45,22 @@
 @implementation WebGLLayer
 
 @synthesize context=_context;
+@synthesize glLayer=_glLayer;
```
I suppose this needs to be conditionalized as well.

JonWBedard committed:
[webkitscmpy] Linkify http urls in pull-requests
https://bugs.webkit.org/show_bug.cgi?id=230655

Reviewed by NOBODY (OOPS!).

* Scripts/libraries/webkitscmpy/webkitscmpy/pull_request.py:
(PullRequest):
(PullRequest.escape_html): Escape any html and linkify http urls.
(PullRequest.unescape_html): Unescape escaped html and strip links from http urls.
(PullRequest.create_body): Use <pre> and escape html in commit messages, unless opted out.
(PullRequest.parse_body): Match both <pre> and ```, escape html if <pre> was used.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/remote/bitbucket.py:
(BitBucket.PRGenerator.create): Opt out of linkifying http in comment body.
(BitBucket.PRGenerator.update): Ditto.
* Scripts/libraries/webkitscmpy/webkitscmpy/test/pull_request_unittest.py:
